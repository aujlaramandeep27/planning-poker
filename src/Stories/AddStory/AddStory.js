import React from 'react';
import {connect} from 'react-redux';
import {viewHost} from '../../Navigation/route-actions';

export function AddStory(props) {

    return (        
        <div className="uk-container uk-text-center@m uk-margin-top uk-margin-bottom">
            <h1 className="uk-heading-small uk-heading-divider">Add Story</h1>
            <a href="#/host">
                <span uk-icon="icon: arrow-left; ratio: 3" className="uk-position-large uk-position-top-left"></span>
            </a>
            <div className="uk-align-center uk-width-1-2@m">      
                <form onSubmit={event => {AddStory(event)}}>
                    <div className="uk-margin">
                        <div className="uk-inline uk-width-1-2">
                            <span className="uk-form-icon" uk-icon="icon: link"></span>
                            <input name="storyInputName" className="uk-input" placeholder="Name" type="text"/>
                        </div>
                        <button className="uk-button uk-button-primary uk-width-1-4">
                            ADD
                        </button>
                    </div>
                </form>
            </div>
        </div>
    );

    function AddStory(event) {
        event.preventDefault();
        props.goToHost();
    }
};

function mapStateToProps(state) {
    return {
    }
};
 
 function mapDispatchToProps(dispatch) {
    return {
        goToHost: () => dispatch(viewHost())
    }
};
 
export default connect(mapStateToProps, mapDispatchToProps)(AddStory)