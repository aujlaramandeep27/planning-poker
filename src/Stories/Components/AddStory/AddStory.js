import React from 'react';
import { connect } from 'react-redux';
import { viewHost}  from '../../../Navigation/route-actions';
import { withHeader } from '../../../Common/Header';

export function AddStory({goToHost}) {

    return (        
        <>
            <div className="uk-align-center uk-width-1-2@m">      
                <form onSubmit={event => {AddStory(event)}}>
                    <div className="uk-margin">
                        <div className="uk-inline uk-width-1-2">
                            <span className="uk-form-icon" uk-icon="icon: link"></span>
                            <input name="storyInputName" className="uk-input" placeholder="Name" type="text"/>
                        </div>
                        <button className="uk-button uk-button-primary uk-width-1-4">
                            ADD
                        </button>
                    </div>
                </form>
            </div>
        </>
    );

    function AddStory(event) {
        event.preventDefault();
        goToHost();
    }
};

function mapStateToProps(state) {
    return {
    }
};
 
 function mapDispatchToProps(dispatch) {
    return {
        goBack: () => dispatch(viewHost()),
        goToHost: () => dispatch(viewHost())
    }
};
 
export default connect(mapStateToProps, mapDispatchToProps)(withHeader(AddStory, 'Add Story'));