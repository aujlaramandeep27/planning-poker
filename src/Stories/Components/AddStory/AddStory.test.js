import {AddStory} from "./AddStory";

it('AddStory renders correctly', () => {
    const props = {
        goToHost: jest.fn(),
        postStory: jest.fn()
    }
    expect(AddStory(props)).toMatchSnapshot();
});