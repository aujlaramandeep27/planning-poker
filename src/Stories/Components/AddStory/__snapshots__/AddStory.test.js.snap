// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStory renders correctly 1`] = `
<div
  className="uk-align-center uk-width-1-1@m"
>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="uk-margin"
    >
      <div
        className="uk-inline uk-width-1-2"
      >
        <span
          className="uk-form-icon"
          uk-icon="icon: link"
        />
        <input
          className="uk-input"
          name="storyInputName"
          placeholder="Name"
          type="text"
        />
      </div>
      <button
        className="uk-button uk-button-primary uk-width-1-4"
      >
        ADD
      </button>
    </div>
  </form>
</div>
`;
